### PR TITLE
add window message notifications for embeds: (fixes #333)

### LINKED ISSUE #333: [Feature]: Additional navigation events for embeds
### Context

Add a way to send events to containing page when a <replay-web-page> embed:
- starts navigating to a new page
- finishes navigating to a new page
- gets a 404 when navigating to a new page.

### What change would you like to see?

This will make integration into the Browsertrix QA system more robust (per webrecorder/browsertrix#1780) and can be used by other tools integrating RWP as well.

### Requirements

_No response_

### Todo

_No response_

### PR DESCRIPTION
**Pull request by ikreymer**
- add 'replayNotFoundError' field to 'urlchange' (propagating 'archive-not-found' from wabac.js) to embed frame, with url/ts, for 404 pages
- propagate 'page-loading' with loading true/false, to indicate when a page started/finished loading

The `page-loading` message does not have the URL that is being loaded, as that is only available when loading is finished (via other messages like urlchange). The 'archive-not-found' message is initially sent from wabac.js to the immediate iframe, and then the 'replayNotFoundError' field is added to the urlchange event.

Only send one urlchange event per URL change (keep track of change).
Also ensure multi timestamp change uses replaceHistory to avoid an extra navigation.

Fixes #333

### CODE REVIEW

**Comment by ikreymer** [comment]
Message sequence for 404 page:
```
{type: 'page-loading', loading: true}
{type: 'page-loading', loading: false}
{type: 'archive-not-found', url: 'https://example.com/', ts: '20221002034558'}
```

For valid page:
```
{type: 'page-loading', loading: true}
{type: 'page-loading', loading: false}
{type: 'urlchange', url: 'https://example.com/', ts: '20221002034558', ...}
```

File: package.json [hunk 1]
```diff
@@ -12,7 +12,7 @@
   "dependencies": {
     "@fortawesome/fontawesome-free": "^5.15.4",
     "@shoelace-style/shoelace": "~2.15.1",
-    "@webrecorder/wabac": "^2.18.3",
+    "@webrecorder/wabac": "github:webrecorder/wabac.js",
```
Comment (SuaYoo) [inline package.json:15]: Should we lock this down with a commit hash?

Comment (ikreymer) [reply package.json:15]: oh, yes, this was just for testing! Will do a 2.18.4 release when all is ready.

**Comment by ikreymer** [comment]
> What do you think about dispatching a `wr-page-loading` event from `Embed`? It may be more straightforward for `<replay-web-page>` embeds, to call like:
> 
> ```
> <replay-web-page
>   replayBase="/js/"
>   source="/wacz/example.wacz"
>   url="https://webrecorder.net"
>   @wr-page-loading=${handlePageLoading}
> ></replay-web-page>
> ```

yeah, that could work, though would be in addition to the postMessage messages (at least initially), since its sending between multiple window frames. I guess could have a `@wr-page-loading` and `@wr-url-changed` maybe?

**Review by SuaYoo — APPROVED** [review]

**Pull request merged**